dbt/tests/data_tests/generic_tests/expect_missingness_between.sql

```diff
@@ -0,0 +1,16 @@
+{% test expect_missingness_between(
+    model,
+    column_name,
+    lower_bound=0,
+    upper_bound=1
+) %}
+
+SELECT
+    COUNT(*) AS total_records,
+    COUNT(*) - COUNT({{ column_name }}) AS null_records,
+    (COUNT(*) - COUNT({{ column_name }})::FLOAT) / NULLIF(COUNT(*), 0) AS null_proportion
+FROM {{ model }}
+HAVING null_proportion < {{ lower_bound }}
+    OR null_proportion > {{ upper_bound }}
```

Comment:
This test could also replace the `not_all_null` test that we already have. But I think that's going to be replaced with the asset checks for accidentally totally null columns.

Reply:
Yeah I feel like a test with a configurable range is more useful than just checking for no nulls at all.